static/js/ReaderPanel.jsx

```diff
@@ -41,21 +41,35 @@ import {
   ToggleSet, InterfaceText, EnglishText, HebrewText, SignUpModal,
 } from './Misc';
 import {ContentText} from "./ContentText";
+import ReaderDisplayOptionsMenu from "./ReaderDisplayOptionsMenu";
+import DropdownMenu from "./common/DropdownMenu";
 
 
 class ReaderPanel extends Component {
   constructor(props) {
     super(props);
 
-    let state = this.clonePanel(props.initialState);
-    state["initialAnalyticsTracked"] = false;
-    state.width = this.props.multiPanel ? 1000 : 500; // Assume we're in a small panel not using multipanel
-    state.backButtonSettings = null;
-    this.state = state;
+    this.state = {
+      ...this.clonePanel(props.initialState),
+      initialAnalyticsTracked: false,
+      width: this.props.multiPanel ? 1000 : 500, // Assume we're in a small panel not using multipanel
+      backButtonSettings: null,
+      data: null,
+    };
     this.sheetRef = React.createRef();
     this.readerContentRef = React.createRef();
   }
+  conditionalSetTextData() {
```

Comment:
i'm not sure what this comment neams. the code here uses Sefaria.getTextFromCurrVersions, i.e. the cache.
but when thinking about it, now that the data is in the context, maybe it will be better to get it from there (but we are using the data on componentDidUpdate, so if we want to use context we'll have to use getSnapshotBeforeUpdate for storing the prev context as snapshot).